File: python/matrix/matrix.py
class Matrix:
    def __init__(self, string):
        self.rows = self._parse(string)
        self.columns = self._transform(self.rows)

    def _parse(self, string):
        rows = string.split('\n')
        return [list(map(int, row.split(' '))) for row in rows]

    def _transform(self, rows):
        size = len(rows[0])
        return [[r[i] for r in rows] for i in range(size)]

File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows, :columns

  def initialize(str)
    @rows = build(str)
    @columns = @rows.transpose
  end

  private

  def build(str)
    str.split("\n").map { |row| row.split(' ').map(&:to_i) }
  end
end
